backend/pkg/analytics/charts/metric_user_journey.go

```diff
@@ -38,732 +36,621 @@ type JourneyData struct {
+		var result []SunburstData = make([]SunburstData, 0)
+		for _, l := range levels {
+			if element.Depth == l.Depth && element.Name == l.Name && element.Type == l.Type {
+				result = append(result, l)
 			}
 		}
-
-		nodeID++
+		return result
 	}
 
-	// Filter out nodes with no connections
-	nodeHasConnection := make(map[int]bool)
-	for _, link := range links {
-		nodeHasConnection[link.Source] = true
-		nodeHasConnection[link.Target] = true
-	}
-
-	// Make sure central nodes are included even if they don't have links
-	for _, node := range nodes {
-		if node.Depth == centralDepth {
-			nodeHasConnection[node.ID] = true
+	var sumValues = func(children []SunburstData) uint64 {
+		var result uint64 = 0
+		for _, c := range children {
+			result += c.Value
 		}
+		return result
 	}
 
-	var filteredNodes []Node
-	for _, node := range nodes {
-		if nodeHasConnection[node.ID] {
-			filteredNodes = append(filteredNodes, node)
+	var response []SunburstData = make([]SunburstData, 0)
+	var levels []SunburstData = make([]SunburstData, 0)
+	var depth uint64 = 1
+	for _, r := range rows {
+		if r.EventNumberInSession > 2 {
+			break
 		}
-	}
-
-	// Reassign IDs to be sequential
-	nodeIDMap := make(map[int]int)
-	var finalNodes []Node = make([]Node, 0, len(filteredNodes))
-
-	for newID, node := range filteredNodes {
-		nodeIDMap[node.ID] = newID
-		node.ID = newID
-		finalNodes = append(finalNodes, node)
-	}
-
-	// Update link references
-	var finalLinks []Link = make([]Link, 0, len(links))
-	for _, link := range links {
-		srcID, srcExists := nodeIDMap[link.Source]
-		tgtID, tgtExists := nodeIDMap[link.Target]
-
-		if srcExists && tgtExists {
-			link.Source = srcID
-			link.Target = tgtID
-			finalLinks = append(finalLinks, link)
+		var children = &response
+		var currentElement = SunburstData{
+			Name:     r.EValue,
+			Type:     r.EventType,
+			Depth:    uint16(r.EventNumberInSession),
+			Value:    0,
+			Children: &[]SunburstData{}}
+		if r.EventNumberInSession > depth {
+			depth += 1
 		}
-	}
 
-	return JourneyData{
-		Nodes: finalNodes,
-		Links: finalLinks,
-	}, nil
-}
-
-func (h *UserJourneyQueryBuilder) buildQuery(p *Payload) (string, error) {
-	// prepare event list filter
-	events := p.MetricValue
-	if len(events) == 0 {
-		events = []string{"LOCATION"}
-	}
-	vals := make([]string, len(events))
-	for i, v := range events {
-		vals[i] = fmt.Sprintf("'%s'", v)
-	}
-	laterCond := fmt.Sprintf("e.\"$event_name\" IN (%s)", strings.Join(vals, ","))
-
-	// build start and exclude conditions
-	startConds, _ := BuildEventConditions(p.StartPoint, BuildConditionsOptions{DefinedColumns: mainColumns, MainTableAlias: "e"})
-	excludeConds, _ := BuildEventConditions(p.Exclude, BuildConditionsOptions{DefinedColumns: mainColumns, MainTableAlias: "e"})
+		var savedElements []SunburstData = getAllFromCurrentLevel(currentElement, levels)
+		if len(savedElements) == 0 {
+			*children = append(*children, currentElement)
```

Comment:
According to my earlier explanation about Go slice internals, could you please remove the unnecessary pointer usage?